lib/kitchen/directions/bake_footnotes/main.rb

```diff
@@ -6,6 +6,41 @@ module BakeFootnotes
       def self.v1(book:)
```

Comment:
There's an additional difference in how footnotes are numbered: instead of 

```ruby
book.chapters.each do |chapter|
  bake_footnotes_within(chapter)
end
```

as in v1, v2 uses

```ruby
bake_footnotes_within(book.chapters)
```

For this difference, I could pass in a variable like `within: :book` or `within: :chapter` and then do something like

```ruby
case within
when :book
  bake_footnotes_within(book.chapters)
when :chapter
  # iteration to bake by chapter
end
```

It was at this point that I started to feel like the divergence could be sufficient for a v2. But v1 is obviously still possible.

Regarding the roman numeral argument, only the top-level page footnotes are numbered with roman numerals, while the footnotes within chapters are numbered normally. Would `number_format_pages` work for the variable name instead, leaving it open to separately specify the chapter numbering with `number_format_chapters` later?